pkg/controllers/jobset_controller.go

```diff
@@ -334,14 +324,17 @@ func (r *JobSetReconciler) suspendJobSet(ctx context.Context, js *jobset.JobSet,
 	})
 }
 
-func (r *JobSetReconciler) resumeJobSetIfNecessary(ctx context.Context, js *jobset.JobSet, ownedJobs *childJobs) error {
+func (r *JobSetReconciler) resumeJobSetIfNecessary(ctx context.Context, js *jobset.JobSet, ownedJobs *childJobs, jobsetSuspended bool) error {
 	log := ctrl.LoggerFrom(ctx)
-	// StartupPolicy suspends existing jobs but does not set a condition for JobSet.
-	// This guards against resumeJobSet resuming all the jobs when using startuppolicy.
-	if !apimeta.IsStatusConditionTrue(js.Status.Conditions, string(jobset.JobSetSuspended)) {
-		return nil
-	}
 
+	// Handle StartupPolicy if jobset is not suspended
+	if !jobsetSuspended {
+		err := r.executeStartupPolicyOnSuspend(ctx, js, ownedJobs)
```

Comment:
I am guessing this change is temporary, and the goal is to merge the logic in executeStartupPolicyOnSuspend into the main loop at line 338 as suggested in https://github.com/kubernetes-sigs/jobset/pull/246#discussion_r1452760904

Reply:
I aimed to address this comment https://github.com/kubernetes-sigs/jobset/pull/246#discussion_r1452761684 but I'll move some things around.

Reply:
Updated.